Documentation/discovery_protocol.md

```diff
@@ -0,0 +1,109 @@
+This document talks about the technical details of discovery service protocol. It helps users to learn about how discovery service works and what behavior to expect.
```

Comment:
It says the same thing as line `The rest of this article will walk...`. I will delete this one.